### ISSUE TITLE
The code of the Mixed Attention Module

### ISSUE THREAD

**Issue by EavanLi** [MCG-NJU/MixFormer#26]
I'm a little confused about the details of the MAM implementation.

In def forward_test() of the class Attention() of lib/models/mixformer/mixformer_online.py, it seems that there is only one calculation process of attention, where q, k, and v are obtained as q = rearrange(search, 'b c h w -> b (h w) c').contiguous(), k = torch.cat([self.t_k, self.ot_k, k], dim=1), and v = torch.cat([self.t_v, self.ot_v, v], dim=1). However, Figure 2 of this paper contains 1 multi-head attention function and 2 attention operations, which cannot directly correspond to the code.

I'm guessing you took a more convenient code in your implementation. Sorry for my poor understanding of code, please explain the details so that I can understand better.

**Comment by yutaocui** [comment]
The operation in Figure. 2 is equal to `forward_test()` & `set_online()`, which perform attention_s (search as query) and attention_t (templates as query) respectively. We only update the template memory every N step.
